scripts/apps/monitoring/controllers/AggregateCtrl.js

```diff
@@ -99,10 +102,42 @@ export function AggregateCtrl($scope, api, desks, workspaces, preferencesService
                                 return {'type': 'workspace', 'groups': []};
                             });
                     } else if (activeWorkspace.type === 'desk') {
-                        var desk = self.deskLookup[activeWorkspace.id];
-                        if (desk && desk.monitoring_settings) {
-                            return {'type': 'desk', 'groups': desk.monitoring_settings};
-                        }
+                        return preferencesService.get(PREFERENCES_KEY)
+                            .then(function(preference) {
+                                var desk = self.deskLookup[activeWorkspace.id];
+
+                                // Read available groups from user preferences first
+                                if (preference && preference[activeWorkspace.id] && preference[activeWorkspace.id].groups) {
+                                    if (desk.monitoring_settings !== undefined) {
+                                        // difference between desk.monitoring_settings and preference[activeWorkspace.id].groups
+                                        var diff = _.xor(_.map(desk.monitoring_settings, '_id'),
+                                             _.map(preference[activeWorkspace.id].groups, '_id'));
+
+                                        if (_.isEmpty(diff)) {
+                                            // update groups in preferences with any changes in desk's monitoring settings groups.
+                                            _.map(preference[activeWorkspace.id].groups, function(group) {
```

Comment:
Generally `map` is used when we need the return value, which is not the case here, so please use native `forEach`. It's best to avoid using `lodash` since modern JS pretty much offers all you need.